File: framework/test2/char/chars1d/Chars1d.java
import soda.unittest.*;

import java.util.*;
import java.util.function.Supplier;
import java.util.stream.*;

import soda.leetcode.*;
import soda.unittest.Validators;
import soda.unittest.work.TestWork;

class Solution {
    public List<Character> doubleList(char[] chars) {
        var res = new ArrayList<Character>();
        for (char ch : chars) {
            res.add(ch);
        }
        for (char ch : chars) {
            res.add(ch);
        }
        return res;
    }
}

public class Chars1d implements Supplier<TestWork> {

    @Override
    public TestWork get() {
        var work = new TestWork(new Solution(), "doubleList");
        // var work = TestWork.forStruct(Struct.class);
        // work.setValidator((e, r) -> {...});
        work.setCompareSerial(true);
        return work;
    }

    public static void main(String[] args) throws Exception {
        new Chars1d().get().run();
    }

}

File: framework/test2/char/chars1d/chars1d.rb
# step [1]: implement solution function
# @param {String[]} chars
# @return {String[]}
def double_list(chars)
  chars + chars
end

if __FILE__ == $0
  require 'soda/unittest/work'
  ns = Soda::Unittest
  # step [2]: setup function/return/arguments
  work = ns::TestWork.create(method(:double_list))
  # work = ns::TestWork.forStruct(CLASS)
  # work.validator = -> (e, r) { e is equal to r }
  work.compare_serial = true
  puts work.run(ARGF.read)
end

